chall12/vde-dios.c

```diff
@@ -0,0 +1,55 @@
+#include <string.h>
+
+void    swap_str(char *str, int start, int finish)
+{
+    char        aux;
+    int         i;
+
+    i = start + 1;
+    while (i < (finish - start + 1) / 2 + start)
+    {
+        aux = str[i];
+        str[i] = str[finish + start - i];
+        str[i] == ')' ? str[i] = '(' : (str[i] == '(' ? str[i] = ')' : str[i]);
+        aux == ')' ? aux = '(' : (aux == '(' ? aux = ')' : aux);
+        str[finish + start - i] = aux;
+        i++;
+    }
+    if ((finish - start + 1) % 2)
+        str[i] == ')' ? str[i] = '(' : (str[i] == '(' ? str[i] = ')' : str[i]);
+}
+
+char *ft_reverse_parenthesis(const char *str)
+{
+    int i, j, level;
+    char *aux = strdup(str);
+
+    i = -1;
+    level = 0;
+    while (str[++i])
+        str[i] == '(' ? level++ : (str[i] == ')' ? level-- : level);
```

Comment:
Using ternaries can help you gaining lines, but be careful not to hurt your code's readability!  
Also, with this syntax check, it seems you're not handling the parentheses being in that order `)    (` !  
Example of string that fails:
```
never close ) before opening (
```

Reply:
I will account for that next time. I didn't feel very confy nesting several ternary but just wanted to try to see how it worked..

About the parenthesis error... There's ever enough tests, I am planning to tatoo it on my skin 😅